pyincore/analyses/multiobjectiveretrofitoptimization/multiobjectiveretrofitoptimization.py

```diff
@@ -0,0 +1,1343 @@
+                              and NCSA IN-CORE Dev Team
+
+        Related publications
+
+
+        Args:
+            incore_client (IncoreClient): Service authentication.
+        """
+
+    # Column descriptors
+    __Q_col = 'Q_t_hat'
+    __Q_rs_col = 'Q_t_hat_rs'
+    __SC_col = 'Sc'
+    __SC_rs_col = 'Sc_rs'
+
+    __budget_default = 0.2
+
+    def __init__(self, incore_client):
+        super(MultiObjectiveRetrofitOptimization, self).__init__(incore_client)
+
+    def run(self):
+        """Execute the multiobjective retrofit optimization analysis using parameters and input data."""
+        # Read parameters
+        model_solver = self.get_parameter('model_solver')
+        num_epsilon_steps = self.get_parameter('num_epsilon_steps')
+
+        budget_available = self.__budget_default
+        if self.get_parameter('max_budget') != 'default':
+            budget_available = self.get_parameter('budget_available')
+
+        inactive_submodels = []
+
+        in_subm = self.get_parameter('inactive_submodels')
+
+        if in_subm is not None:
+            inactive_submodels = in_subm
+
+        # Perform code scaling
+        scaling_factor = 1.0
+        if self.get_parameter('scale_data'):
+            scaling_factor = self.get_parameter('scaling_factor')
+
+        building_repairs_csv = self.get_input_dataset('building_repairs_data').get_dataframe_from_csv()
+        strategy_costs_csv = self.get_input_dataset('strategy_costs_data').get_dataframe_from_csv()
+
+        # Convert Z columns to text in both datasets
+        building_repairs_csv['Z'] = building_repairs_csv['Z'].astype(str)
+        strategy_costs_csv['Z'] = strategy_costs_csv['Z'].astype(str)
+
+        self.multiobjective_retrofit_optimization_model(model_solver, num_epsilon_steps, budget_available,
+                                                        scaling_factor, inactive_submodels, building_repairs_csv,
+                                                        strategy_costs_csv)
+
+    def multiobjective_retrofit_optimization_model(self, model_solver, num_epsilon_steps, budget_available,
+                                                   scaling_factor, inactive_submodels, building_functionality_csv,
+                                                   strategy_costs_csv):
+        """Performs the computation of the model.
+
+        Args:
+            model_solver (str): model solver to use for analysis
+            num_epsilon_steps (int): number of epsilon values for the multistep optimization algorithm
+            budget_available (float): budget constraint of the optimization analysis
+            scaling_factor (float): scaling factor for Q and Sc matrices
+            inactive_submodels (list): submodels to avoid during the computation
+            building_functionality_csv (pd.DataFrame): building repairs after a disaster event
+            strategy_costs_csv (pd.DataFrame): strategy cost data per building
+        """
```

Comment:
space, we use Google style, see https://sphinxcontrib-napoleon.readthedocs.io/en/latest/example_google.html